### DISCUSSION TITLE
External Caching for Self-Hosted Runners on Google Cloud

### DISCUSSION THREAD

**Discussion by cznovluk** [community/community#132156] [Packages]
### External Caching for Self-Hosted Runners on Google Cloud

We've been using the archived GitHub repository [gcs-cache-action](https://github.com/MansaGroup/gcs-cache-action) to handle caching for our self-hosted runners in Google Cloud. This allows us to use the same caching commands in our pipeline as we would with native GitHub runners.

Given that this repository is archived, we are looking for any better solutions to manage external caching with self-hosted runners. If there are no superior alternatives, would GitHub consider officially taking over this repository to ensure its continued maintenance and support?

**Comment by SajanGhimire1** [comment]
@cznovluk 
you can replace gcs-cache-action by using Google Cloud Storage directly. Implement caching by scripting uploads and downloads with gsutil in your CI/CD pipelines. This method gives you full control over caching without depending on external actions.

**Reply by cznovluk** [reply]
Thank you for your response. 👍  I will definitely give it a try. Do you have any guide or instructions? I couldn’t find anything clear.

**Reply by SajanGhimire1** [reply]
> Thank you for your response. 👍 I will definitely give it a try. Do you have any guide or instructions? I couldn’t find anything clear.

You can set up caching for your Google Cloud runners by using gsutil from Google Cloud Storage, check out this guide: [gsutil installation & usage]```https://cloud.google.com/storage/docs/gsutil_install```. will help you integrate caching into your CI/CD workflows effectively.
